include/devicetree/clocks.h

```diff
@@ -54,6 +54,39 @@ extern "C" {
 #define DT_CLOCKS_LABEL_BY_IDX(node_id, idx) \
 	DT_PROP_BY_PHANDLE_IDX(node_id, clocks, idx, label)
 
+/**
+ * @brief Get clock controller "name" (label property) by name
+ *
+ * It's an error if the clock controller referenced by the phandle
+ * in property "clocks" at index "idx" has no label property.
+ *
+ * Example devicetree fragment:
+ *
+ *     clk1: clkctrl@... {
+ *             label = "CLK_1";
+ *     };
+ *
+ *     clk2: clkctrl@... {
+ *             label = "CLK_2";
+ *     };
+ *
+ *     n: node {
+ *             clocks = <&clk1 10 20>, <&clk2 30 40>;
+ *             clock-names = "alpha", "beta";
+ *     };
+ *
+ * Example usage:
+ *
+ *     DT_CLOCKS_LABEL_BY_NAME(DT_NODELABEL(n), beta) // "CLK_2"
+ *
+ * @param node_id node identifier
+ * @param name lowercase-and-underscores "clock" name
+ * @return the label property for the referenced node by name
+ * @see DT_PHANDLE_BY_NAME()
+ */
+#define DT_CLOCKS_LABEL_BY_NAME(node_id, name) \
+	DT_PROP(DT_PHANDLE_BY_NAME(node_id, clocks, name), label)
```

Comment:
You can just use DT_LABEL(DT_PHANDLE_BY_NAME(...)) if you prefer.

Reply:
Was matching what we had in dma.h and adc.h.

Reply:
Ok by me to defer to future cleanup